src/match/queue.test.ts

```diff
@@ -0,0 +1,51 @@
+import { Queue } from './queue';
+
+const queue = new Queue();
+describe('get the queue', () => {
+  it('should return an empty queue', () => {
+    expect(queue.get()).toEqual([]);
+  });
+
+  it('should return all queue members', () => {
+    queue.join('test1');
+    const date = queue.get()[0]?.joinAt;
+    expect(queue.get()).toEqual([
+      {
+        id: 'test1',
+        joinAt: date,
+      },
+    ]);
+  });
+
+  it('should return first 2 members', () => {
+    queue.join('test2');
+    queue.join('test3');
+    const result = queue.getFirstNItems(2);
+    const date1 = result[0]?.joinAt;
+    const date2 = result[0]?.joinAt;
```

Comment:
I think we can leave out the `joinAt` field. It won't be used in the logic to determine who will be matched, and if we need to add it in, we can add it in later. Keeping `joinAt` here will contribute to more code pollution and extra stuff in these unit tests that aren't actually being used.

Reply:
but in the **ranked** match making, we need to consider the join time and expend the range of Elo score, is that right?

Reply:
We'll leave that for future work :). For now let's not worry about this!